### DISCUSSION TITLE
parquet support for java sdk.

### DISCUSSION THREAD

**Discussion by lecssmi** [minio/minio#14291] [Q&A] [answered]
Is it possible to directly read some columns of parquet using s3's java sdk?
It seems that aside from the select api, minio doesn't support partial reading of files.

Configured according to the tips in the [article,](https://docs.min.io/docs/minio-select-api-quickstart-guide.html) but it has no effect. Does it only work with select api?

**Comment by harshavardhana** [comment, marked as answer]
Parquet is not fully supported @lecssmi https://docs.min.io/docs/minio-select-api-quickstart-guide.html

**Reply by lecssmi** [reply]
Thanks,  if I only need to  read some columns in a parquet file, will I  still need to read the entire file ?

**Reply by klauspost** [reply]
`select column1,column2 from s3object` will return only "column1" and "column2".

**Reply by lecssmi** [reply]
> 

I use the s3-java-sdk that implements the HFDS interface to read, does it work?
